client/rest-high-level/src/test/java/org/elasticsearch/client/ClusterClientIT.java

```diff
@@ -303,6 +303,8 @@ public void testClusterHealthNotFoundIndex() throws IOException {
         assertThat(response.status(), equalTo(RestStatus.REQUEST_TIMEOUT));
         assertThat(response.getStatus(), equalTo(ClusterHealthStatus.RED));
         assertNoIndices(response);
+        assertWarnings("HTTP status code for an internal cluster health timeout will be changed from 408 to 200 in 8.0.0. " +
+            "Set the `es.cluster_health.request_timeout_200` property to `true` to opt in for it.");
```

Comment:
Why "internal" here? Also let's just say "a future version" in case the breaking change doesn't land by 8.0.

```suggestion
        assertWarnings("The HTTP status code for a cluster health timeout will be changed from 408 to 200 in a future version. " +
            "Set the `es.cluster_health.request_timeout_200` system property to `true` to suppress this message and opt in to the future behaviour now.");
```

(probably needs reformatting for line length, sorry there's no easy way to check for that in Github)